### ISSUE TITLE
Fix the issue in PulseAudio itself or supply patch in Ubuntu

### ISSUE THREAD

**Issue by bensteUEM** [edrose/dell-dock-audio-fix#2]
Hi there, great patch, thanks for taking the time to look into this issue.
I've opened an issue on launchpad in order to get this sorted in Ubuntu too.

https://bugs.launchpad.net/ubuntu/+source/pulseaudio/+bug/1745765

**Comment by edrose** [comment]
I don't believe the issue is in PulseAudio since the files I got to fix it are from the PulseAudio source repo. The Launchpad bug you submitted has been marked as Duplicate and it looks like a fix is nearly ready for Ubuntu.

I've added a bug report to Arch as well.

https://bugs.archlinux.org/task/57266

**Comment by superm1** [comment]
It's probably worth referring to the commit in your readme.
https://github.com/pulseaudio/pulseaudio/commit/60c0edd5286dbb731c671ad3e6886c1e3e1eb067

**Comment by edrose** [comment]
@superm1 Done - thanks for pointing that out!

**Comment by gdestuynder** [comment]
pulseaudio fixed it
that said, this repo comes in the top matches when you look for dell usb audio dock issues, and a recent kernel introduced a somewhat similar issue by renaming the dock in alsa.. which breaks alsa-lib..
the fix is to reflect: https://patchwork.kernel.org/patch/10647179/ in /usr/share/alsa/ucm/Dell-WD15-Dock/HiFi.conf

**Comment by edrose** [comment]
@gdestuynder that's good to know. I haven't played around with getting the dock to work since installing this when I made it so I was unaware.

Are you suggesting I should edit this repo to mark it obsolete/update it to do the new fix? I'm happy to make some changes when I get time if it's needed

**Comment by gdestuynder** [comment]
i think this repo is "obsolete" now yes, though it's good to keep it around for reference.

i'm also saying that there's another issue that started coming up recently, which will match similar terms, for which you can make a similar patch for it if you want, or other people might just run into my comment and do it manully (including my future self if this isn't fixed upstream, when I forget which file I have changed 3 month from now :)

**Comment by edrose** [comment]
Okay, I'll make some changes to the readme tonight to reflect this and link people to the appropriate place for the new bug

**Comment by grrodre** [comment]
Hello everyone and sorry to reopen this old bug.

I have a WD19 docking station and i am facing the same problem.

> Debian GNU/Linux bullseye/sid \n \l
> Linux xps 5.5.0-1-amd64 #1 SMP Debian 5.5.13-2 (2020-03-30) x86_64 GNU/Linux

pactl list cards

> Card #1
>         Name: alsa_card.usb-Generic_USB_Audio_200901010001-00
>         Driver: module-alsa-card.c
>         Owner Module: 25
>         Properties:
>                 alsa.card = "2"
>                 alsa.card_name = "WD19 Dock"
>                 alsa.long_card_name = "Dell-WD15-Dock"
>                 alsa.driver_name = "snd_usb_audio"
>                 device.bus_path = "pci-0000:00:14.0-usb-0:9.3.4:1.0"
>                 sysfs.path = "/devices/pci0000:00/0000:00:14.0/usb3/3-9/3-9.3/3-9.3.4/3-9.3.4:1.0/sound/card2"
>                 udev.id = "usb-Generic_USB_Audio_200901010001-00"
>                 device.bus = "usb"
>                 device.vendor.id = "0bda"
>                 device.vendor.name = "Realtek Semiconductor Corp."
>                 device.product.id = "402e"
>                 device.product.name = "USB Audio"
>                 device.serial = "Generic_USB_Audio_200901010001"
>                 device.string = "2"
>                 device.description = "USB Audio"
>                 module-udev-detect.discovered = "1"
>                 device.icon_name = "audio-card-usb"
>         Profiles:
>                 input:analog-stereo: Analog Stereo Input (sinks: 0, sources: 1, priority: 65, available: yes)
>                 output:analog-stereo: Analog Stereo Output (sinks: 1, sources: 0, priority: 6500, available: yes)
>                 output:analog-stereo+input:analog-stereo: Analog Stereo Duplex (sinks: 1, sources: 1, priority: 6565, available: yes)
>                 off: Off (sinks: 0, sources: 0, priority: 0, available: yes)
>         Active Profile: output:analog-stereo+input:analog-stereo
>         Ports:
>                 analog-input-mic: Microphone (priority: 8700, latency offset: 0 usec)
>                         Properties:
>                                 device.icon_name = "audio-input-microphone"
>                         Part of profile(s): input:analog-stereo, output:analog-stereo+input:analog-stereo
>                 analog-output-headphones: Headphones (priority: 9900, latency offset: 0 usec)
>                         Properties:
>                                 device.icon_name = "audio-headphones"
>                         Part of profile(s): output:analog-stereo, output:analog-stereo+input:analog-stereo


http://alsa-project.org/db/?f=616054544b8748431c354f1affbfd705c88fad37

If i run manualy speaker-test and select the proper subdevice, i get sound over the lineoutput. I cannot set it as default (or i dont know how to do it).
It seems that if i am able to set the UCM2 configuration from WD15 should work. Any advise?

Thank you in advance.

**Comment by superm1** [comment]
It's entirely possible there was a regression in upstream, perhaps during the timeframe that stuff switched from UCM -> UCM2.  I would recommend bringing this to the alsa mailing lists.

**Comment by grrodre** [comment]
Thank you for the quick response and the suggestion Mario

**Comment by PieterJanVanAeken** [comment]
@grrodre did you ever bring this to the mailing list by any chance? I am running into the same issue at the moment, where only the front headphone jack of my WD19-TB is correctly functioning.

**Comment by grrodre** [comment]
@PieterJanVanAeken Hello Peter. I subscribed to the list and sent a message. I did not get any response (or i did not find it) and received a lot of messages from the list so I unsubscribed. 
I decided to use the headphone instead of the lineout hoping that someday it will be fixed :).

If you contact to the list or get any news let me know!

Greets 
Greg.

**Comment by elfrinjo** [comment]
With the latest Kernel, PA and ALSA this seems to be fixed in Arch Linux

**Comment by grrodre** [comment]
Hi everyone, with debian testing it seems it has been fixed.

Regards
